### DISCUSSION TITLE
Getting url from server

### DISCUSSION THREAD

**Discussion by cardinalgeo** [Kitware/trame#353] [General] [answered]
Hi Trame Team :) I'm trying to get the url (i.e., specifying the port) for a trame app in order to create an iframe. For example, [here](https://github.com/Kitware/trame-iframe/blob/ec30fff3126d35d6473cd136d49f776f36fa4b36/examples/trame-trame/parent.py#L9) in the `trame-trame` example (`trame-iframe` repo), the child url is defined explicitly rather than pulled programmatically from the child server. How can I instead get it from the server directly? I suspect the solution may be quite trivial, but I've done a good bit of digging through the trame repos without any luck!

Best, Robert

**Comment by jourdain** [comment, marked as answer]
Well, it depends on the details of your setup. Is trame server running within Jupyter? Or using a launcher? Where is that trame server, and how does it get started?

Assuming you have your hand on the server that you want to connect to, you can get its port by doing `server.port` once running...

**Reply by cardinalgeo** [reply]
Ultimately it'd be within jupyter, though I'm generally playing with embedding trame in other apps (e.g., in the same vein as [this](https://discourse.holoviz.org/t/trame-support-in-panel-to-fully-support-vtk/5275/9) thread). 

Thanks for the comment on `server.port`. I had observed it used [here](https://github.com/Kitware/trame/blob/e447803c39192dff554b128178eadaa66db3f981/trame/app/jupyter.py#L30) but I guess my question deals with the other parameters used to construct the url — in what cases does the url actually deviate from the format `"http://localhost:1236/"`? I could insert the port number into a string of that format but, for example, in what situations is the `ui` parameter used?

**Reply by jourdain** [reply]
If it is within Jupyter, that is another story, as we have built-in stuff to streamline things such as ipywidget integration and if the trame-jupyter-extension is used, then the URL will be different as no other port or network connection will be needed.

You should look at the following links:
- https://github.com/Kitware/trame/blob/master/examples/jupyter/multi-tasks.ipynb
- https://github.com/Kitware/trame-client/blob/master/trame_client/ui/core.py#L19-L52

**Reply by cardinalgeo** [reply]
Thanks Seb — I'll take a look at those and repost here if I run into more issues. For now, I'll accept your previous comment re: ports as the answer!
